roachtest: enable multiple_active_portals in ORM tests

rafiss opened the pull request:
As part of this, some tests were changed to use a non-root user, so that
they can pick up the new setting.

This makes it so we can configure default values for session variables
more easily in many of the tests.

fixes: https://github.com/cockroachdb/cockroach/issues/98641

Also updates a few of the blocklists to account for new tests:
fixes https://github.com/cockroachdb/cockroach/issues/118294
fixes https://github.com/cockroachdb/cockroach/issues/118303
fixes https://github.com/cockroachdb/cockroach/issues/118306

Release note: None

rafiss committed:
roachtest: enable multiple_active_portals in ORM tests

As part of this, some tests were changed to use a non-root user, so that
they can pick up the new setting.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/114621)
<!-- Reviewable:end -->

rafiss committed:
roachtest: use non-root user in ORM tests

This makes it so we can configure default values for session variables
more easily in many of the tests.

Release note: None

rafiss commented:
kicked off a run here https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel?buildTypeTab=overview&branch=114621&mode=builds

fqazi reviewed (approved):
<img class="emoji" title=":lgtm_strong:" alt=":lgtm_strong:" align="absmiddle" src="https://reviewable.io/lgtm_strong.png" height="20" width="61"/> 

> Reviewed 4 of 4 files at r1, 16 of 16 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/114621#-:-Np5VGcz7Mw16xqCI2ZS:b-uwkowd)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @annrpom, @DarrylWong, and @renatolabs)



<!-- Sent from Reviewable.io -->

rafiss commented:
tftr!

bors r+
